### DISCUSSION TITLE
Z_OFFSET is not saving the calibration.

### DISCUSSION THREAD

**Discussion by jimjworkshop** [Guilouz/Klipper-Flsun-Speeder-Pad#91] [General]
I have installed the latest updates and config files. I am trying to calibrate the Z Offset. I am using the Z_OFFSET_CALIBRATION macro. I have the sensor attached and it comes down and touches the build plate a couple of times. Then raises up[ a little so I can remove the sensor. The using the manual adjustment screen that pops up I lower the print head down to the surface and use a piece of paper to set the height. I click ACCEPT and then the SAVE macro. I then start a print and the print head starts printing at the same height it was when the sensor was attached. What am I doing wrong?

**Comment by chuong8899** [comment]
im having the same issue. Did you fix it?

**Comment by AlexRJ30** [comment]
Tha same for me too.

**Comment by javaandy** [comment]
Migrated yesterday and have the same problem. The printer is printing in the air.

**Comment by AlexRJ30** [comment]
I got round the problem by playing with the baby step when printing.
Once you're satisfied, you cancel the print and it's automatically saved as a z-offset.

**Reply by javaandy** [reply]
Thanks. Something was completely messed up on my config. I reset everything and next followed your suggestion. It's looking good now.

**Comment by Guilouz** [comment]
Calibration must be done like this :

- Mesure Probe Offset (Z-Offset) with Z_OFFSET_CALIBRATION macro and paper and save, this will have the action of defining the z offset in the printer.cfg file.
- Start Endstop Calibration with ENDSTOPS_CALIBRATION macro.
- Start Delta Calibration with DELTA_CALIBRATION macro.
- Start Bed Leveling with BED_LEVELING macro.
- Optional: you can also apply a 2mm Safety Offset with SECURITY_OFFSET macro. This will have the action of defining a gcode offset of 2mm above the measured value during the Probe Offset, to avoid scratching the plate if the measured value is not perfect.

When it's done start a print and check your first layer by adjusting the babysteps. And it is this value (the difference from the actual Z Offset) that will be saved in the variables.cfg file.

Above all, do not save the value of the Z-Offset to probe or to endstops in Mainsail, this will affect Delta calibration measurements. Thanks to the variables.cfg file, the value defined during the babysteps (gcode offset) is automatically reloaded when Klipper starts.
